### PR TITLE
CDAP-4121 remove Id from ArtifactRange

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-RUT302

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java [hunk 1]
```diff
@@ -759,15 +759,15 @@ private NamespaceId validateAndGetScopedNamespace(NamespaceId namespace, Artifac
         try {
           range = ArtifactRange.parse(parent);
           // only support extending an artifact that is in the same namespace, or system namespace
-          if (!range.getNamespace().equals(Id.Namespace.SYSTEM) &&
-            !range.getNamespace().equals(namespace.toId())) {
+          if (!range.getNamespace().equals(NamespaceId.SYSTEM) &&
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java]: any chance ``range.getNamespace()`` can be null? If so, reverse this?

File: cdap-common/src/main/java/co/cask/cdap/common/conf/ArtifactConfigReader.java [hunk 2]
```diff
@@ -82,8 +83,8 @@ public ArtifactConfig read(Id.Namespace namespace, File configFile) throws IOExc
 
         // check namespaces in parents are either system or the specified namespace
         for (ArtifactRange parent : config.getParents()) {
-          Id.Namespace parentNamespace = parent.getNamespace();
-          if (!parentNamespace.equals(Id.Namespace.SYSTEM) && !parent.getNamespace().equals(namespace)) {
+          NamespaceId parentNamespace = parent.getNamespace();
+          if (!parentNamespace.equals(NamespaceId.SYSTEM) && !parent.getNamespace().equals(namespace.toEntityId())) {
```
Comment (bdmogal) [inline cdap-common/src/main/java/co/cask/cdap/common/conf/ArtifactConfigReader.java]: same comment here

**Review by bdmogal — APPROVED** [review]
Couple minor nits, rest LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java (hunk 1 above)
Comment (albertshau) [reply cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java]: it shouldn't ever be null, but always safer to reverse. will do.

**Pull request merged**